cvmfs/catalog_traversal.h

```diff
@@ -147,26 +104,70 @@ class ObjectFetcher;
  *       as unlinking of the catalog database file.
  *
  *
- * @param CatalogT        the catalog class that should be used for traversal
- *                        usually this will be either catalog::Catalog or
- *                        catalog::WritableCatalog
- *                        (please consider to use the typedef'ed versions of
- *                        this template i.e. [Readable|Writable]CatalogTraversal)
- *
- * @param ObjectFetcherT  Strategy Pattern implementation that is supposed to be
- *                        used mainly for unit-testability. Normal usage should
- *                        not bring up the need to actually set this to anything
- *                        different than the default value (see below)
+ * @param ObjectFetcherT  Strategy Pattern implementation that defines how to
+ *                        retrieve catalogs from various backend storage types.
+ *                        Furthermore the ObjectFetcherT::catalog_t is the type
+ *                        of catalog to be instantiated by CatalogTraversal<>.
  *
- * CAUTION: the Catalog* pointer passed into the callback becomes invalid
+ * CAUTION: the catalog_t* pointer passed into the callback becomes invalid
  *          directly after the callback method returns, unless you create the
  *          CatalogTraversal object with no_close = true.
  */
-template<class CatalogT, class ObjectFetcherT = ObjectFetcher>
-class CatalogTraversal : public Observable<CatalogTraversalData<CatalogT> > {
+template<class ObjectFetcherT>
+class CatalogTraversal
+  : public Observable<CatalogTraversalData<typename ObjectFetcherT::catalog_t> >
+{
  public:
-  typedef CatalogTraversalData<CatalogT> CallbackData;
-  typedef CatalogT                       Catalog;
+  typedef ObjectFetcherT                      object_fetcher_t;
```

Comment:
I find this confusing because `some_name_t` looks like something from a C library.  I like the idea of being able to see a derived type but please think about a way of naming them in camel case, like it is done for template parameters.